front/app/modules/commercial/insights/hooks/useInsightsInputs.test.ts

```diff
@@ -0,0 +1,152 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import useInsightsInputs, { QueryParameters } from './useInsightsInputs';
+import { Observable, Subscription } from 'rxjs';
+import { delay } from 'rxjs/operators';
+import { waitFor } from 'utils/testUtils/rtl';
+import { insightsInputsStream } from 'modules/commercial/insights/services/insightsInputs';
+
+const viewId = '1';
+
+const queryParameters: QueryParameters = {
+  category: '3',
+  pageSize: 10,
+  pageNumber: 12,
+  search: 'search',
+  sort: '-approval',
+};
+
+const mockInputs = {
+  data: {
+    currentPage: 1,
+    lastPage: 2,
+    list: [
+      {
+        id: '4e9ac1f1-6928-45e9-9ac9-313e86ad636f',
+        type: 'input',
+        relationships: {
+          source: {
+            data: {
+              id: '4e9ac1f1-6928-45e9-9ac9-313e86ad636f',
+              type: 'idea',
+            },
+          },
+          categories: {
+            data: [
+              {
+                id: '94a649b5-23fe-4d47-9165-9beceef2dcad',
+                type: 'category',
+              },
+              {
+                id: '94a649b5-23fe-4d47-9165-9becedfg45sd',
+                type: 'category',
+              },
+            ],
+          },
+          suggested_categories: {
+            data: [],
+          },
+        },
+      },
+      {
+        id: '54438f73-12f4-4b16-84f3-a55bd118de7e',
+        type: 'input',
+        relationships: {
+          source: {
+            data: {
+              id: '54438f73-12f4-4b16-84f3-a55bd118de7e',
+              type: 'idea',
+            },
+          },
+          categories: {
+            data: [],
+          },
+          suggested_categories: {
+            data: [],
+          },
+        },
+      },
+    ],
+  },
+};
+
+let mockObservable = new Observable((subscriber) => {
+  subscriber.next(mockInputs);
+}).pipe(delay(1));
+
+jest.mock('modules/commercial/insights/services/insightsInputs', () => {
+  return {
+    insightsInputsStream: jest.fn(() => {
+      return {
+        observable: mockObservable,
+      };
+    }),
+  };
+});
+
+describe('useInsightsInputs', () => {
+  it('should call useInsightsInputs with correct default arguments', async () => {
+    renderHook(() => useInsightsInputs(viewId));
+    expect(insightsInputsStream).toHaveBeenCalledWith(viewId, {
+      queryParameters: {
+        category: undefined,
+        'page[number]': 1,
+        'page[size]': 20,
+        search: undefined,
+        sort: 'approval',
+      },
+    });
+  });
+  it('should call useInsightsInputs with correct non-default arguments', async () => {
```

Comment:
Could it be useful to test that it picks up changes to the props too or am I overzealous and we can trust react and ourselves on this one ?

Reply:
Yes, really good idea! Helped me catch that we didn't include pageSize in the useEffect dependencies. We are not using it, so it wasn't a huge issue, but it's anyway good to catch these things :)

Reply:
We are currently not changing page size but it is an option in the useEffect. If we ever add it to the pagination - agreed we should reset the pageNumber as well.